store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/utils/TableUtils.java

```diff
@@ -216,7 +221,63 @@ public static Connector getConnector(final String instanceName, final String zoo
         try {
             return instance.getConnector(userName, new PasswordToken(password));
         } catch (final AccumuloException | AccumuloSecurityException e) {
-            throw new StoreException("Failed to create accumulo connection", e);
+            throw new StoreException("Failed to create accumulo connection (using password-based authentication)", e);
+        }
+    }
+
+    /**
+     * Creates a connection (using Kerberos authentication) to
+     * an accumulo instance using the provided parameters
+     *
+     * @param instanceName the instance name
+     * @param zookeepers   the zoo keepers
+     * @param principal    the Kerberos principal
+     * @param keytabPath   the path to the Kerberos keytab
+     * @return A connection to an accumulo instance
+     * @throws StoreException failure to create an accumulo connection
+     */
+    public static Connector getConnectorKerberos(final String instanceName, final String zookeepers,
+                                                 final String principal, final String keytabPath) throws StoreException {
+        final Instance instance = new ZooKeeperInstance(instanceName, zookeepers);
+        try {
+            // Configure Hadoop UGI to use Kerberos (if it's not already)
+            if (!UserGroupInformation.isSecurityEnabled()) {
+                Configuration conf = new Configuration();
+                conf.set("hadoop.security.authentication", "kerberos");
+                conf.set("hadoop.security.authorization", "true");
+                UserGroupInformation.setConfiguration(conf);
+            }
+            // If already logged in with Keytab, then check if ticket needs renewal, else do initial login
```

Comment:
Is a connection made for every Operation or just when the graph is created?

Reply:
It is done for every operation. However the Kerberos renewal/login is only done as often as is required. If the connection was only created once then there would be no easy way to ensure it did not expire. This isn't an issue for user/pass which cannot expire like this.